### PR TITLE
DAOS-7675 vos: Remove object cache

### PR DESCRIPTION
**Pull request by jolivier23**
It doesn't add much value in the best case
In the worst case, where you have many objects, it degrades
performance significantly so let's simplify the code and
remove it.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: src/vos/vos_obj_cache.c [hunk 1]
```diff
@@ -228,45 +67,15 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
...
-		D_GOTO(failed_2, rc);
-
-	obj = container_of(lret, struct vos_object, obj_llink);
-
-	if (obj->obj_zombie)
-		D_GOTO(failed, rc = -DER_AGAIN);
-
-	if (intent == DAOS_INTENT_KILL) {
-		if (vos_obj_refcount(obj) > 2)
-			D_GOTO(failed, rc = -DER_BUSY);
```
Comment (NiuYawei) [inline src/vos/vos_obj_cache.c:246]: Seems these 'refcount' & 'zombie' was used for preventing an object being deleted by vos_obj_delete() while it's still holding by someone else, is there any equivalent in this patch?

Comment (jolivier23) [reply src/vos/vos_obj_cache.c:246]: Yeah, this is probably a little too simplistic.   @mchaarawi gets really good IO500 results with this patch compared to master branch which is why I've cleaned it up a bit to upload.   I wonder if we can leverage the LRU timestamp cache.  I'll poke around a bit.

**Comment by Nasf-Fan** [comment]
It is some interesting about how to use the DRAM cache against PMEM based storage. If the object in the cache is never reused, then maintaining object cache is some overhead. But why we introduced object cache at the beginning? I guess it was related with performance consideration. Anyway, all that depend on the IO modes. So whether @mchaarawi 's good IO500 results is common case or some special test that similar as the case of never reusing the same object in the test? Just want to make clear.

On the other hand, Niu's question about how to prevent busy object being destroyed, that mainly caused by CPU yield during some VOS operation, such as fetch and enumeration. If we cannot remove the yield, then we still need some kind of busy objects cache. If the concurrent busy-yield operation count is not too large, then simple d_list_t attaching to the container may be enough.

**Comment by gnailzenh** [comment]
As we discussed, before we do this, we should just improve the LRU cache and see if it can resolve the performance issue.

**Pull request closed without merging**